content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+    environment { /* <1> */
+        CC = 'clang'
+    }
+    stages {
+        stage('Example') {
+            environment { /* <2> */
+                AN_ACCESS_KEY = credentials('my-prefined-secret-text') /* <3> */
+            }
+            steps {
+                sh 'printenv'
+            }
+        }
+    }
+}
+----
+<1> An `environment` directive used in the top-level `pipeline` block will
+apply to all steps within the Pipeline.
+<2> An `environment` directive defined within a `stage` will only apply the
+given environment variables to steps within the `stage`.
+<3> The `environment` block has a helper method `credentials()` defined which
+can be used to access pre-defined Credentials by their identifier in the
+Jenkins environment.
+
+==== options
+
+The `options` directive allows configuring Pipeline-specific options from
+within the Pipeline itself. Pipeline provides a number of these options, such
+as `buildDiscarder`, but they may also be provided by plugins, such as
+`timestamps`.
+
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
+|===
+
+===== Available Options
+
+buildDiscarder:: Persist artifacts and console output for the specific number
+of recent Pipeline runs. For example: `options { buildDiscarder(logRotator(numToKeepStr: '1')) }`
+disableConcurrentBuilds:: Disallow concurrent executions of the Pipeline. Can
+be useful for preventing simultaneous accesses to shared resources, etc. For
+example: `options { disableConcurrentBuilds() }`
+skipDefaultCheckout:: Skip checking out code from source control by default in
+the `agent` directive. For example: `options { skipDefaultCheckout() }`
+timeout:: Set a timeout period for the Pipeline run, after which Jenkins should
+abort the Pipeline. For example: `options { timeout(time: 1, unit: 'HOUR') }`
+retry:: On failure, retry the entire Pipeline the specified number of times.
+For example: `options { retry(3) }`
+timestamps:: Prepend all console output generated by the Pipelien run with the
+time at which the line was emitted. For example: `options { timestamps() }`
+
+[[options-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    options {
+        timeout(time: 1, unit: 'HOUR') /* <1> */
+    }
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+<1> Specifying a global execution timeout of one hour, after which Jenkins will
+abort the Pipeline run.
+
+[NOTE]
+====
+A comprehensive list of available options is pending the completion of
+link:https://issues.jenkins-ci.org/browse/INFRA-1053[INFRA-1503].
+====
+
+==== parameters
+
+The `parameters` directive provides a list of parameters which a user should
+provide when triggering the Pipeline. The values for these user-specified
+parameters are made available off of the `params` object within the Pipeline,
```

Comment:
"made available off of the params object within the Pipeline" -> "made available to Pipeline steps via the `params` object"